app/views/challenge/gift_exchange/_challenge_signups.html.erb

```diff
@@ -1,6 +1,20 @@
-<h2 class="heading"><%= ts("Sign-ups for %{collection}", :collection => @collection.title) %></h2>
-  
+<h2 class="heading">
+  <% if @query %>
+    <%= search_header @challenge_signups, nil, "Signup" %>
+  <% else %>
+    <%= ts("Sign-ups for %{collection}", collection: @collection.title) %>
+  <% end %>
+</h2>
+
 <ul class="navigation actions" role="menu">
+  <li class="search" role="search">
+      <%= form_tag collection_signups_path(@collection), class: "simple search", method: :get do |form| %>
```

Comment:
This is really small and not blocking: looks like 4-space tabs, and the submit tag is not translation method'ed. (Although I'm not sure anymore _how_ we're doing translation. >.>)